### DISCUSSION TITLE
Regression on including informative alt text on banner images

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [renovatebot/renovate#27393] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Back in #25714 I'd reported unclear alt text for the banner images in PRs:

> Proposal: can we modify the alt text on that image to something just a bit more informative? Say:
>
> > ```diff
> > - > [![Mend Renovate](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)
> > + > [![Mend Renovate logo banner](https://app.renovatebot.com/images/banner.svg)](https://renovatebot.com)
> > ```

That was fixed the next day (which was amazing!). But it looks like the fix for #25714 was regressed/reverted by accident? Using https://github.com/danvk/gravlax/pull/57 as an example:

> ```html
> <img src="..." alt="Mend Renovate" data-canonical-src="https://app.renovatebot.com/images/banner.svg" style="max-width: 100%;">
> ```

Could the more informative alt text be added back please?

### Relevant debug logs

n/a

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by viceice** [comment]
@Gabriel-Ladzaretti can you have a look?

**Comment by HonkingGoose** [comment]
Hi @JoshuaKGoldberg! :wave: 

Thank you for telling us about this!

I recently improved the alt text for our readme and our published docs. So I did a bit of reading about the best practices for alt text. Our docs and readme have the Renovate banner logo image as well, so I needed to figure out the best alt text. :wink: 

[WebAIM, Alternative Text, Logos](https://webaim.org/techniques/alttext/#logos) says:

> Many websites link the main branding logo to the home page. Providing alternative text for the image, such as your company name (`alt="Acme Company"`), will usually suffice. The word "logo" (`alt="Acme Company Logo"`) is usually not an important part of the image's content or function. Similarly, it is usually not necessary to indicate that the image links to the home page (`alt="Acme Company home page"`) because this is a common convention—hearing "Link, Graphic, Acme Company" at the top of a web page is sufficient for a screen reader user to know that it is a logo linked to the home page.

The way I'm reading this: a short text like `Mend Renovate` is better than `Mend Renovate banner logo`.

I went with `Renovate banner` for the repository readme and the docs homepage banner images. To be consistent, we could go with `Mend Renovate banner` for the Renovate PR's from the official Mend-hosted app?

Let me know what you think!

Greetings, @HonkingGoose

**Reply by JoshuaKGoldberg** [reply, marked as answer]
Ah great, awesome that you're doing a deep dive on this!

I'm not an expert here, so I can't confidently speak to whether short text like `Mend Renovate` is right. It sounds like it to me but a regular accessibility professional (at least one of whom I would hope Mend has some relationship with) should really chime in.

But from a layperson's perspective, that sounds perfect. 😄

**Reply by HonkingGoose** [reply]
# Summary

This is basically a really deep dive into the general concept of accessibility, Voice Over usage, alt text, normal text, the important of good writing and good structure, and accessibility advocacy.

Feel free to only respond to parts of my essay! :wink: 

## Deep dive intro

> Ah great, awesome that you're doing a deep dive on this!

I had a lot of fun, _and_ frustration, trying to figure out how to use the Voice Over feature of iOS on the Renovate docs, and on the repository.

### Voice Over modes

Voice Over is a really different way to get information, that feels very random at first. Voice Over has different modes that you can select with a "virtual rotor" like:

- container
- headings
- links
- character
- forms
- point at thing with finger

### Example of using Voice Over's modes to navigate

So you might go into _headings_ mode, to quickly skim the headings, and find the thing you need. 

Or you can use _link_ mode to get a quick overview of the clickable things on a page.

You can also point at things on the screen with your finger to get a call out of what's under your finger. So you can use your finger as "eye".

And with character mode you can get a character-by-character spelling of a word.

You can also let Voice Over read you the whole page from top to bottom, and pause/rewind as needed by swiping left/right and giving certain taps to pause/start.

### About skip to content invisible link

Most pages have a invisible-to-eyes "skip to main content" link. When you're using a screen-reader that is vitally important to quickly get to the main content.

### I recommend you try Voice Over (or a similar tool)

You should try Voice Over, or a similar tool too! It was really enlightening to experience first-hand how it works, and how it feels to use.

#### _Before_ you enable Voice Over

_Before_ you enable Voice Over make sure you:

- Know how to get back out of the special mode!
- Read Apple's documentation for Voice Over, especially the different swipe and tap commands.
- Watch a video that explains how Voice Over works.
- Remember that you need to tap to confirm an action, first you will hear for example, something like: "Link, Renovate docs, button", the _button_ part means you can tap the screen to select the button or follow the link.
- Do _not_ expect any important messages/calls when trying Voice Over, as answering them might be a bit hard to figure out under pressure.
- Enable the _Caption Panel_ option to get subtitles for the Voice Over speech output: Settings -> Accessibility -> Voice Over -> scroll down to the _Caption Panel_ option, enable it.

##### Extra tips

- You can control how slow or fast Voice Over talks at you. I recommend using the default middle position, at least for now.
- You can download a better voice for Voice Over: go to Accessibility -> Voice Over -> Select the Speech menu -> Select Voice -> Select a Voice from the list -> Select the _Speaker name (Enhanced)_ voice to download the best quality voice.

## About alt text

> I'm not an expert here, so I can't confidently speak to whether short text like `Mend Renovate` is right. (snipped)

I think that for decorative items like logo's, shorter is better than long. Conversely for important images, we should be _descriptive_, but still as short as possible. I got really tired from listing to the computer generated speech after a short while.

### Short text helps many users

That also applies to the length and complexity of the documentation itself! For example a text with good heading, short sentences, easy words, good descriptive text, and good link names helps:

- blind users who have to listen to computer generated speech
- users who get distracted easily (ADHD, ADD)
- those for who English is a second language
- those who struggle with difficult words and long sentences
- not yet disabled people who just want to get things working quickly

## Feedback from accessibility professional

> It sounds like it to me but a regular accessibility professional (at least one of whom I would hope Mend has some relationship with) should really chime in.

I'm deaf, and have other disabilities I do not want to mention here. So where I can, I push for accessibility. :smile: 

Of course I can only talk from experience about my own disabilities, from my perspective. More viewpoints would certainly help. :smile: 

## Conclusion

Accessibility is very important to me. I'd like for you to try Voice Over or a screen-reader too! And I welcome new viewpoints to help Mend and the Renovate project improve the accessibility.

If you've read this far, thank you very much! :heart:
